.github/workflows/build_ncs.yml

```diff
@@ -0,0 +1,64 @@
+name: Build binary for Nordic NCS
+
+# Runs when pushed to main, semver tagged release or any PR
+on:
+  push:
+    branches: [ main ]
+    tags: [ 'v*.*.*' ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    # Uses matrix to test combinations. Update when SDK changes or to add new boards.
+    strategy:
+      matrix:
+        ZEPHYR_SDK_VERSION: [0.16.1]
+        BOARD: ["nrf9160dk_nrf9160_ns","aludel_mini_v1_sparkfun9160_ns"]
+
+    steps:
+      - name: Install OS dependencies
+        run: >
+          sudo apt-get install -y
+          device-tree-compiler
+          git
+          ninja-build
+          python3
+          python3-pip
+          python3-wheel
+          wget
+          xz-utils
+
+      - name: Install build tools
+        run: |
+          pip3 install wheel
+          pip3 install cmake west
+
+      - name: Install Zephyr SDK
+        run: |
+          wget -q "https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v${{ matrix.ZEPHYR_SDK_VERSION }}/zephyr-sdk-${{ matrix.ZEPHYR_SDK_VERSION }}_linux-x86_64_minimal.tar.xz"
+          mkdir -p /opt/zephyr-sdk
+          tar -xvf zephyr-sdk-${{ matrix.ZEPHYR_SDK_VERSION }}_linux-x86_64_minimal.tar.xz -C /opt/zephyr-sdk --strip-components=1
+          /opt/zephyr-sdk/setup.sh -t arm-zephyr-eabi
+          rm zephyr-sdk-${{ matrix.ZEPHYR_SDK_VERSION }}_linux-x86_64_minimal.tar.xz
+
+      - name: Build with West
+        run: |
+          west init -m https://github.com/$GITHUB_REPOSITORY .
+          west update
+          west zephyr-export
+          pip3 install -r deps/zephyr/scripts/requirements.txt
+          west build -p -b ${{ matrix.BOARD }} app -- -DCONFIG_MCUBOOT_IMAGE_VERSION=\"0.0.0\"
+          mv build/zephyr/zephyr.bin build/zephyr/${{ matrix.BOARD }}.bin
+          mv build/zephyr/zephyr.elf build/zephyr/${{ matrix.BOARD }}.elf
```

Comment:
Sorry, one more thing: the two types of binaries most important to us are:

* merged.hex (initial flash to the board using nRF Connect Desktop programmer tools)
* app_update.bin (used for OTA updates)

Can this be updated to include these files with the following naming scheme?

* golioth-reference-design-template_v`version`_`boardname`_full.hex
* golioth-reference-design-template_v`version`_`boardname`_update.bin

Reply:
To clarify, do you want to remove `.bin` and `.elf` as well?

And when you say `version`, is that implying a release, so not including the current behavior of PRs or pushes to main?

Reply:
I haven't used the .elf files for flashing. I've only built for nRF9160-based boards (aludel and DK) which output:
* `build/zephyr/merged.hex`
* `build/zephyr/app_update.bin`
For releases I've renamed these based on the board and the revision number (which I've used as the release number as well)
![image](https://github.com/golioth/reference-design-template/assets/367685/78e79ecb-dbe7-4fb8-b38a-0f48ac01b5aa)

> And when you say version, is that implying a release, so not including the current behavior of PRs or pushes to main?

Yes, PRs and merges are guaranteed to be well-tested. Only releases have that guarantee. So I'm hesitant to make binaries available for those. But we have a testing checklist for cutting releases so we can be fairly certain the binaries will work well if used.

Do you have a use case in mind for binaries for every PR?

Reply:
> Do you have a use case in mind for binaries for every PR?

The biggest benefit I see at this point for automated builds in PRs is simply verifying that the build completes without errors on not-my-machine before the release is created.

Reply:
I think build testing is great, but we don't need add the binaries as artifacts.

Reply:
Made it `golioth-{{repo_name}}-{{board_name}}_{{file}}_{{short_git_hash}}.*`. Since we're not doing release automation yet, I didn't/couldn't include version. Until we do, you can do a quick rename to add version.